sharding/indexHashedNodesCoordinator.go

```diff
@@ -306,6 +306,14 @@ func (ihgs *indexHashedNodesCoordinator) GetValidatorWithPublicKey(
 		}
 	}
 
+	for shardId, shardWaiting := range nodesConfig.waitingMap {
```

Comment:
Isn't this time consuming? This will be used in the upcoming network sharding feature branch as a way to know the shard ID of the public key provided. It will be called each time a connection is made (currently about 5 connections/sec are done)

Reply:
fixed by using a map for caching